### DISCUSSION TITLE
How to determine the ratio of master node and agent node

### DISCUSSION THREAD

**Discussion by 983183947** [k3s-io/k3s#8082] [Q&A] [answered]
How to determine the ratio of master node and agent node? If I have a cluster with 20 nodes and all of them are master nodes, will there be any problems?

**Comment by brandond** [comment]
Don't do that. You shouldn't ever have more than 5, max 7, etcd nodes - as per the etcd documentation. For the control-plane, most components are leader-elected so there's not much value in having more than 2 servers. Find the number of server nodes that meets your quorum requirements, and scale them up. Scale out agent nodes to meet your workload requirements.

**Reply by 983183947** [reply]
Thanks Reply! I use an external mysql database. Will too many masters affect the load of mysql?

**Reply by brandond** [reply, marked as answer]
That would have been a good thing to mention up front.

Yes, additional servers will put additional load on the database for no substantial gain. Same as with etcd, I probably wouldn't run more than 2 or 3 masters.
